expertise/create_dataset.py

```diff
@@ -294,11 +313,22 @@ def get_submissions(self):
         submissions = []
 
         for invitation_id in invitation_ids:
-            submissions.extend(list(openreview.tools.iterget_notes(
-                self.openreview_client, invitation=invitation_id)))
+            submissions_v1 = list(openreview.tools.iterget_notes(
+                self.openreview_client, invitation=invitation_id))
+
+            if not submissions_v1:
+                submissions.extend(list(openreview.tools.iterget_notes(
+                    self.openreview_client_v2, invitation=invitation_id)))
+            else:
```

Comment:
Don't need this `else`, just extend it before the `if`, even if it's empty, it won't matter.

Reply:
Do you know what the behavior of the `openreview-py` client is when attempting to fetch notes that don't exist? I'm fine with this change if a failed `get_notes` and `get_note` call results in something like empty lists.

The current behavior of the mock client in the tests returns `None` in this situation, which breaks the tests because the following line fails:
https://github.com/openreview/openreview-expertise/blob/60b9a077f3bb16a03cc639e23d3aae261741d254/expertise/create_dataset.py#L313

But I can adjust the mock client to be more accurate to the live Python client if I know what happens when trying to get a note/notes that don't exist.

Reply:
No, but I assumed it'd return an empty list same as when you use `openreview_client_v2`. @melisabok can you confirm this?

Reply:
get_notes return a list
get_note return a Note object or an exception if the note doesn't exist

you can use tools.get_note that returns a None if the note doesn't exist.

when querying the notes by invitation we should assume the result will come from one of the APIs.